lib/src/main/kotlin/com/swmansion/starknet/signer/StarkCurveSigner.kt

```diff
@@ -16,7 +16,11 @@ class StarkCurveSigner(private val privateKey: Felt) : Signer {
     override val publicKey: Felt by lazy { StarknetCurve.getPublicKey(privateKey) }
 
     override fun signTransaction(transaction: Transaction): Signature {
-        return StarknetCurve.sign(privateKey, transaction.hash).toList()
+        if (transaction.hash == null) {
+            throw IllegalArgumentException("Transaction hash must be set before signing")
```

Comment:
Perhaps we could improve the error message here? This implies that this is something user should do, and this is probably not a case, right?